### ISSUE TITLE
profile() only accepts properties from UserProfileProperties

### ISSUE THREAD

**Issue by YakQin** [gitbrent/SpRestLib#40]
To implement the enhancement in #38 you have made changes to sprLib.user(options).profile(). Now it seems we cannot provide any Properties from Person Properties like 'DirectReports', 'DisplayName' etc

sprLib.user(options).profile('DirectReports') -->this is not working, error:  DirectReports"`DirectReports` property does not exist in SP.UserProfiles.PeopleManager.UserProfileProperties"

sprLib.user(options).profile('Department')-->this is working fine, 'Department' is part of the UserProfileProperties

**Comment by YakQin** [comment]
i saw this has been fixed in the latest (26th Dec 2018) build

**Comment by YakQin** [comment]
if the property is an array like below ones, we cannot put it in like profile('DirectReports') 

DirectReports
ExtendedManagers
ExtendedReports
Peers

error:
ExtendedManagers"`ExtendedManagers` property does not exist in SP.UserProfiles.PeopleManager.UserProfileProperties"

**Comment by gitbrent** [comment]
Hi @YakQin 

I see what you're saying, and it did indeed used to work differently.

However, when a UserProfile property name is provided, it's used in a specific query to `UserProfile`:
```
SP.UserProfiles.PeopleManager/GetUserProfilePropertyFor(propertyname='SomeProp')
```

Therefore when using another top-level property name (ex: 'DirectReports') zero returns are returned - as it doesn't exist.

So the fix for your issue would be that the code would have to be aware of which ProfileProperties are "genuine" values so we could query all UserProfileProperty values and filter the one being searched for, which is an ugly solution.

Long story short, I'd prefer users to pull all profiles, then select the `[key]` they want if it's not a real UserProfileProperty... e.g.: everything not highlighted:

![screen shot 2019-02-03 at 17 06 53](https://user-images.githubusercontent.com/7218970/52184241-54297b00-27d6-11e9-881f-3d0c4d804efe.png)

**Comment by YakQin** [comment]
noted, thanks a lot
